src/trade_ui.cpp

```diff
@@ -112,6 +113,16 @@ trade_ui::trade_ui( party_t &you, npc &trader, currency_t cost, std::string titl
 {
     _panes[_you]->add_character_items( you );
     _panes[_you]->add_nearby_items( 1 );
+    map &here = get_map();
+    for( const wrapped_vehicle &wv : here.get_vehicles() ) {
+        if( wv.v->is_owned_by( you ) ) {
+            for( const tripoint pos : wv.v->get_points() ) {
+                if( here.inbounds( pos ) && square_dist( you.pos(), pos ) > 1 ) {
+                    _panes[_you]->add_vehicle_items( pos );
+                }
+            }
+        }
+    }
```

Comment:
This adds the items even if the vehicle is on the other side of the map or inaccessible from your current position. I suggest limiting to `PICKUP_RANGE` (=6 tiles) and reusing `add_nearby_items()` with some new bools so you'd have either
```c++
    _panes[_you]->add_nearby_items( 1, true, false ); // map-only items
    _panes[_you]->add_nearby_items( PICKUP_RANGE, false, true ); // veh-only items
```
or just use the legacy trader logic and add everything in range
```c++
    _panes[_you]->add_nearby_items( PICKUP_RANGE );
```
(I hate this one)